src/components/password-modal/password-modal-component.tsx

```diff
@@ -391,21 +397,25 @@ export class PasswordModalComponent extends React.Component<
 
     @bind
     private async updateState(data: { password?: string; biometryAuthResult?: boolean }) {
-        let isPasswordValid = await this.verifyPassword(data.password);
-        if (data?.biometryAuthResult === true) {
-            isPasswordValid = data.biometryAuthResult;
-        }
-
         switch (this.state.currentStep) {
             // Enter PIN Flow
             case ScreenStep.ENTER_PIN:
+                let password = data.password;
+                const shouldConsiderBiometric =
+                    data.password === '' ? this.props.biometricActive : false;
+                if (shouldConsiderBiometric) {
+                    this.props.setDisplayPasswordModal(false);
+                    password = await getPinCode();
+                    this.props.setDisplayPasswordModal(true);
```

Comment:
Maybe you could try to set `this.props.setDisplayPasswordModal(true);` on `onModalHide`, but you need to make sure that it's executed after `app.tsx`. Otherwise, because the flag will be reverted to `true`, `app.tsx` will ask for pin.